nemo/core/classes/modelPT.py

```diff
@@ -1112,15 +1061,93 @@ def teardown(self, stage: str):
 
         super().teardown(stage)
 
+    @classmethod
+    def extract_state_dict_from(cls, restore_path: str, save_dir: str, split_by_module: bool = False):
+        """
```

Comment:
Interesting method... somehow I wasn't aware of it before, good that you have polished it a bit ;)

Reply:
In your defense, I wrote this method and then forgot it existed 💯

Reply:
eheheheh, that's the ultimate! ;)